test_prim_inplace_copy_fwd_nvfuser_cuda_bfloat16 is flaky

See [https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=200582&view=log[…]-5b16-9b83-c0a5a25d03f0&t=97651ec4-0b0f-5455-bbb5-3c30427a0a7e](https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=200582&view=logs&j=5b0799f7-725e-5b16-9b83-c0a5a25d03f0&t=97651ec4-0b0f-5455-bbb5-3c30427a0a7e))

cc @tfogal

Have to say this looks more like numerics... I'll double check the kernel just to be sure about it.
```
    @instantiate()
    def test_prim_inplace_copy_fwd(executor, device, dtype):
        def torch_foo(x, y):
            z = x * y
            z = z + z
            z = x + z
            o = x.copy_(z)
            return o
    
        def foo(x, y):
            z = x * y
            z = z + z
            z = x + z
            # NOTE: nvfuserex doesn't support `return z`, i.e. the copy_from argument
            o = thunder.core.prims.copy_(z, x)
            return o
    
        traced_nvfuser_foo = executor.make_callable(foo)
    
        tdtype = ttorch.to_torch_dtype(dtype)
        a = make_tensor((4, 4), device=device, dtype=tdtype, requires_grad=False)
        b = make_tensor((4, 4), device=device, dtype=tdtype, requires_grad=False)
        a1 = a.detach().clone()
        b1 = b.detach().clone()
        thunder_result = traced_nvfuser_foo(a, b)
        torch_result = torch_foo(a1, b1)
    
        custom_comparator = (
            partial(assert_close, atol=1e-2, rtol=1e-2)
            if dtype in (datatypes.bfloat16, datatypes.float16)
            else assert_close
        )
>       custom_comparator(thunder_result, torch_result)
E       AssertionError: Tensor-likes are not close!
E       
E       Mismatched elements: 1 / 16 (6.2%)
E       Greatest absolute difference: 0.013671875 at index (2, 3) (up to 0.01 allowed)
E       Greatest relative difference: 0.048583984375 at index (2, 3) (up to 0.01 allowed)

thunder/tests/test_inplace_copy.py:44: AssertionError

```

--- a/thunder/tests/test_inplace_copy.py
+++ b/thunder/tests/test_inplace_copy.py
@@ -12,16 +12,12 @@ from thunder.tests.framework import instantiate
 @instantiate()
 def test_prim_inplace_copy_fwd(executor, device, dtype):
     def torch_foo(x, y):
-        z = x * y
-        z = z + z
-        z = x + z
+        z = x + y
         o = x.copy_(z)
         return o
 
     def foo(x, y):
-        z = x * y
-        z = z + z
-        z = x + z
+        z = x + y
         # NOTE: nvfuserex doesn't support `return z`, i.e. the copy_from argument
         o = thunder.core.prims.copy_(z, x)
         return o
@@ -36,14 +32,8 @@ def test_prim_inplace_copy_fwd(executor, device, dtype):
     thunder_result = traced_nvfuser_foo(a, b)
     torch_result = torch_foo(a1, b1)
 
-    # see https://github.com/Lightning-AI/lightning-thunder/issues/295
-    custom_comparator = (
-        partial(assert_close, atol=5e-2, rtol=5e-2)
-        if dtype in (datatypes.bfloat16, datatypes.float16)
-        else assert_close
-    )
-    custom_comparator(thunder_result, torch_result)
-    custom_comparator(a, a1)
+    assert_close(thunder_result, torch_result)
+    assert_close(a, a1)
 
 
 @instantiate(dtypes=(datatypes.floating,))

